### PR TITLE
Resolve Errors for Destruction Warlock

### PR DESCRIPTION
**Pull request by omegabiscuit**
Followed instructions listed [here](https://github.com/WoWAnalyzer/WoWAnalyzer/wiki/Enabling-a-Spec-for-Dragonflight) to fix thrown errors for Destro Warlock

- Imported TALENTS list for spells that were converted to talents in DF
- Fixed JS issues that threw errors for SoulShardTracker

### CODE REVIEW

File: src/analysis/retail/warlock/destruction/modules/core/SpellUsable.ts [hunk 1]
```diff
@@ -34,15 +34,15 @@ class SpellUsable extends CoreSpellUsable {
...
   constructor(options: Options) {
     super(options);
-    this.hasSB = this.selectedCombatant.hasTalent(SPELLS.SHADOWBURN_TALENT.id);
-    this.hasSF = this.selectedCombatant.hasTalent(SPELLS.SOUL_FIRE_TALENT.id);
+    this.hasSB = this.selectedCombatant.hasTalent(TALENTS.SHADOWBURN_TALENT.id);
+    this.hasSF = this.selectedCombatant.hasTalent(TALENTS.SOUL_FIRE_TALENT.id);
     this.addEventListener(Events.SpendResource.by(SELECTED_PLAYER), this.onSpendResource);
     this.addEventListener(
-      Events.applydebuff.by(SELECTED_PLAYER).spell(SPELLS.SHADOWBURN_TALENT),
+      Events.applydebuff.by(SELECTED_PLAYER).spell(TALENTS.INFERNO_TALENT),
```
Comment (emallson) [inline src/analysis/retail/warlock/destruction/modules/core/SpellUsable.ts]: looks like you had another search-and-replace error with Shadowburn -> Inferno

File: src/analysis/retail/warlock/destruction/modules/core/SpellUsable.ts [hunk 2]
```diff
@@ -52,10 +52,10 @@ class SpellUsable extends CoreSpellUsable {
     if (!this.hasSF) {
       return;
     }
-    if (this.isOnCooldown(SPELLS.SOUL_FIRE_TALENT.id)) {
+    if (this.isOnCooldown(TALENTS.INFERNO_TALENT.id)) {
```
Comment (emallson) [inline src/analysis/retail/warlock/destruction/modules/core/SpellUsable.ts]: same, but Soul Fire -> Inferno

File: src/analysis/retail/warlock/destruction/modules/features/Backdraft.tsx [hunk 3]
```diff
@@ -36,8 +37,12 @@ class Backdraft extends Analyzer {
...
   constructor(options: Options) {
     super(options);
-    this._maxStacks = this.selectedCombatant.hasTalent(SPELLS.FLASHOVER_TALENT.id) ? 4 : 2;
-    this._stacksPerApplication = this.selectedCombatant.hasTalent(SPELLS.FLASHOVER_TALENT.id)
+    this._maxStacks = this.selectedCombatant.hasTalent(TALENTS.IMPROVED_CONFLAGRATE_TALENT.id)
+      ? 4
+      : 2;
+    this._stacksPerApplication = this.selectedCombatant.hasTalent(
+      TALENTS.IMPROVED_CONFLAGRATE_TALENT.id,
+    )
```
Comment (emallson) [inline src/analysis/retail/warlock/destruction/modules/features/Backdraft.tsx]: doesn't necessarily need a fix right now bc this stage is more about making it compile than anything else, but this talent appears to no longer increase Backdraft stacks.

File: src/analysis/retail/warlock/destruction/modules/soulshards/SoulShardDetails.tsx [hunk 4]
```diff
@@ -9,7 +10,6 @@ import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
 import Statistic from 'parser/ui/Statistic';
 import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
 
-import 'analysis/retail/warlock/SoulShardDetails.css';
```
Comment (emallson) [inline src/analysis/retail/warlock/destruction/modules/soulshards/SoulShardDetails.tsx:12]: are you certain that this css import is unused? I don't see the css file itself deleted.

File: src/analysis/retail/warlock/destruction/modules/soulshards/SoulShardTracker.ts [hunk 5]
```diff
@@ -201,11 +202,12 @@ class SoulShardTracker extends ResourceTracker {
     spellId: number,
     gain: number,
     waste: number,
-    resource?: ClassResources,
     timestamp?: number,
+    resource?: ClassResources,
   ) {
     const beforeBuilder = this.current % 10;
-    super._applyBuilder(spellId, gain, waste, timestamp, resource);
+    const time = timestamp == null ? -1 : timestamp;
```
Comment (emallson) [inline src/analysis/retail/warlock/destruction/modules/soulshards/SoulShardTracker.ts]: use ?? instead:
```suggestion
    const time = timestamp ?? -1;
```

**Review by emallson — CHANGES_REQUESTED** [review]
mostly looks okay, but again it looks like there are search + replace issues on a couple of talents. I didn't annotate every location that the search + replace applied, but it looks like every use of Shadowburn or Soul Fire needs to be revisited.

File: src/analysis/retail/warlock/destruction/modules/soulshards/SoulShardDetails.tsx (hunk 4 above)
Comment (omegabiscuit) [reply src/analysis/retail/warlock/destruction/modules/soulshards/SoulShardDetails.tsx:12]: I don't think  this file currently exists? I don't see it in the directory, and I get this error when I don't remove it
**Module not found: Error: Can't resolve 'analysis/retail/warlock/SoulShardDetails.css'**

File: src/analysis/retail/warlock/destruction/modules/features/Backdraft.tsx (hunk 3 above)
Comment (omegabiscuit) [reply src/analysis/retail/warlock/destruction/modules/features/Backdraft.tsx]: Yep you're right. I tested in PTR to ensure that's the case

File: src/analysis/retail/warlock/destruction/modules/soulshards/SoulShardDetails.tsx (hunk 4 above)
Comment (emallson) [reply src/analysis/retail/warlock/destruction/modules/soulshards/SoulShardDetails.tsx:12]: The other warlock PR removed it I think. Disregard then

**Review by emallson — APPROVED** [review]

**Pull request merged**